refactor: Replace float based CSS with flex based CSS for primary content.

sagarb90 opened the pull request:
### Description
- Replace float based CSS with flex based CSS for primary content, also added backward compatibility for the same.
- https://trello.com/c/eCydK421/1051-5-content-primary-secondary-flex-based

### Types of changes
- Improvements

### How has this been tested?
<!-- Please describe in detail how you tested your changes. -->

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included any necessary tests <!-- if applicable -->
- [x] I've included developer documentation <!-- if applicable -->
- [x] I've added proper labels to this pull request <!-- if applicable -->

premanshup commented:
@sagarb90 In HTML Markup, the CSS Class is not appearing properly, it is breaking the markup attribute. 

Screenshot: [https://d.pr/i/IU3Vfz](https://d.pr/i/IU3Vfz)

Please verify all the markup once and make sure that we do not leave such breaking markup in the HTML.

premanshup commented on assets/css/unminified/frontend.css:
```diff
@@ -307,17 +287,6 @@ th {
-  clear: both;
-}
-
 .ast-col-xs-1, .ast-col-xs-2, .ast-col-xs-3, .ast-col-xs-4, .ast-col-xs-5, .ast-col-xs-6, .ast-col-xs-7, .ast-col-xs-8, .ast-col-xs-9, .ast-col-xs-10, .ast-col-xs-11, .ast-col-xs-12, .ast-col-sm-1, .ast-col-sm-2, .ast-col-sm-3, .ast-col-sm-4, .ast-col-sm-5, .ast-col-sm-6, .ast-col-sm-7, .ast-col-sm-8, .ast-col-sm-9, .ast-col-sm-10, .ast-col-sm-11, .ast-col-sm-12, .ast-col-md-1, .ast-col-md-2, .as ...
```
@sagarb90 Can we reduce the classes here and add only those which are used?

sagarb90 replied:
@premanshup As you already know these CSS are added from scss using for loop and we already discuss this, so we can not remove and we have already removed these CSS from main.css file (i.e flex based changes).

premanshup replied:
@sagarb90 Okay!

premanshup commented:
@sagarb90 Can we remove before CSS [https://d.pr/i/dpgUlS](https://d.pr/i/dpgUlS)

premanshup commented:
@sagarb90 It looks like we have multiple breakpoints like 922 and 921. I think we should combine such CSS and add it on only a single breakpoint.

Screenshot - [https://d.pr/i/Y3Sq6E](https://d.pr/i/Y3Sq6E)

cc: @Nikschavan

premanshup commented:
@sagarb90 If we are adding `max-width: 1240px` then I think we can remove the `max-width: 100%` CSS. 

Screenshot - [https://d.pr/i/eqacwS](https://d.pr/i/eqacwS)

Please let me know your thoughts.

premanshup commented:
@sagarb90 Can we remove this CSS as well? Let's go through individual HTML for all the layouts and discuss which CSS we can remove.

Screenshot: [https://d.pr/i/BABCzO](https://d.pr/i/BABCzO)

premanshup commented on inc/dynamic-css/container-layouts.php:
```diff
@@ -0,0 +1,130 @@
+/**
+ * Container Layout - Dynamic CSS
+ *
+ * @package astra-builder
```
```suggestion
 * @package astra
```

premanshup commented on inc/dynamic-css/container-layouts.php:
```diff
@@ -0,0 +1,130 @@
+   
+	$page_container_css = '';
+   
+	if ( 'page-builder' == $container_layout ) {
```
@sagarb90 Use ternary operator here.

sagarb90 commented on assets/css/unminified/frontend.css:
```diff
@@ -270,28 +270,8 @@ th {
-@media (min-width: 1200px) {
-  .ast-container {
-    max-width: 100%;
-  }
```
@premanshup Yes these CSS are already added in dynamic CSS Which is added by Navanath and I have also confirm same from him so Removing duplicated CSS - https://share.getcloudapp.com/rRukDBXd.

sagarb90 reviewed (commented):
> @sagarb90 If we are adding `max-width: 1240px` then I think we can remove the `max-width: 100%` CSS.
> 
> Screenshot - https://d.pr/i/eqacwS
> 
> Please let me know your thoughts.

@premanshup We can remove but 1240px; added from dynamic CSS i.e customizer Container layout width If this value will be black then by default it will be apply 100% from static CSS. Please let me know you thought on this scenario.

sagarb90 commented:
> @sagarb90 Can we remove before CSS https://d.pr/i/dpgUlS

@premanshup Yes we have already removed from main.css i.e.flex base changes - https://share.getcloudapp.com/qGuXbbzr
Looks like you are checking frontend.css i.e existing user css.

sagarb90 commented:
> @sagarb90 Can we remove this CSS as well? Let's go through individual HTML for all the layouts and discuss which CSS we can remove.
> 
> Screenshot: https://d.pr/i/BABCzO

@premanshup Yes I have check and removed css you mentioned in the screenshot and also check rest of layout CSS their is no such CSS for  any HTML markup.

sagarb90 commented:
> @sagarb90 It looks like we have multiple breakpoints like 922 and 921. I think we should combine such CSS and add it on only a single breakpoint.
> 
> Screenshot - https://d.pr/i/Y3Sq6E
> 
> cc: @Nikschavan

@premanshup Yes As these are tablet breakpoints so I have added condition for  921 i.e. 100% ,if site container width is empty/ null then apply 100% width CSS. Please let me know your thought on this.

sagarb90 commented:
> @sagarb90 In HTML Markup, the CSS Class is not appearing properly, it is breaking the markup attribute.
> 
> Screenshot: https://d.pr/i/IU3Vfz
> 
> Please verify all the markup once and make sure that we do not leave such breaking markup in the HTML.

@premanshup Yes thank you for checking this I have resolved this now.

vrundakansara commented on inc/dynamic-css/container-layouts.php:
```diff
@@ -0,0 +1,130 @@
+<?php
```
Instead of a new file, can we add this function to `inc/extras.php` file @premanshup 
And call the same function above.

vrundakansara replied:
Alright. 👍🏼

premanshup commented:
> > @sagarb90 Can we remove before CSS https://d.pr/i/dpgUlS
> 
> @premanshup Yes we have already removed from main.css i.e.flex base changes - https://share.getcloudapp.com/qGuXbbzr
> Looks like you are checking frontend.css i.e existing user css.

Yep! I missed the file name, my bad. Looking good 👍🏼

premanshup commented:
> > @sagarb90 If we are adding `max-width: 1240px` then I think we can remove the `max-width: 100%` CSS.
> > Screenshot - https://d.pr/i/eqacwS
> > Please let me know your thoughts.
> 
> @premanshup We can remove but 1240px; added from dynamic CSS i.e customizer Container layout width If this value will be black then by default it will be apply 100% from static CSS. Please let me know you thought on this scenario.

@sagarb90 SO can we load this conditionally? If the value is null or empty from option only then add 100% CSS. Let me know your thoughts?

sagarb90 commented:
> > > @sagarb90 If we are adding `max-width: 1240px` then I think we can remove the `max-width: 100%` CSS.
> > > Screenshot - https://d.pr/i/eqacwS
> > > Please let me know your thoughts.
> > 
> > 
> > @premanshup We can remove but 1240px; added from dynamic CSS i.e from customizer Container layout width, If this value will be blank then by default it will be apply 100% from static CSS. Please let me know you thought on this scenario.
> 
> @sagarb90 SO can we load this conditionally? If the value is null or empty from option only then add 100% CSS. Let me know your thoughts?



> > > @sagarb90 If we are adding `max-width: 1240px` then I think we can remove the `max-width: 100%` CSS.
> > > Screenshot - https://d.pr/i/eqacwS
> > > Please let me know your thoughts.
> > 
> > 
> > @premanshup We can remove but 1240px; added from dynamic CSS i.e from customizer Container layout width, If this value will be blank then by default it will be apply 100% from static CSS. Please let me know you thought on this scenario.
> 
> @sagarb90 SO can we load this conditionally? If the value is null or empty from option only then add 100% CSS. Let me know your thoughts?

@premanshup Yes I have added condition as you mentioned - https://share.getcloudapp.com/E0u9v1jd

vrundakansara commented on inc/class-astra-dynamic-css.php:
```diff
@@ -577,6 +577,8 @@ public static function return_output( $dynamic_css, $dynamic_css_filtered = '' )
 				$parse_css .= astra_parse_css( $old_header_mobile_toggle );
 			}
 
+			$parse_css .= require_once ASTRA_THEME_DIR . 'inc/dynamic-css/container-layouts.php'; // PHPCS:ignore WPThemeReview.CoreFunctionality.FileInclude.FileIncludeFound	
```
@premanshup - Please confirm if this is the correct way to get dynamic CSS?
May be calling a function would be fine here instead of requiring an entire file.

sagarb90 replied:
@vrundakansara Yes you are correct that is better approach and I have changed that to calling function instead require file.

premanshup replied:
@vrundakansara Agree!

premanshup commented:
> SO can we load this conditionally? If the value is null or empty from option only then add 100% CSS. Let me know your thoughts?

@sagarb90  Thank you!
